plugin/Gui/AdminPageApiSettings.php

```diff
@@ -92,23 +92,22 @@ private function addFormModelAPI()
 		$pInputModelApiSecret->setIsPassword(true);
 		$optionNameSecret = $pInputModelApiSecret->getIdentifier();
 		$apiSecret = get_option($optionNameSecret);
-		if (defined('ONOFFICE_CREDENTIALS_ENC_KEY')) {
-			try {
-				$apiSecretDecrypt = $this->_encrypter->decrypt(get_option($optionNameSecret), ONOFFICE_CREDENTIALS_ENC_KEY);
-			} catch (\RuntimeException $e) {
-				$apiSecretDecrypt = $apiSecret;
-			}
-			$apiSecret = $apiSecretDecrypt;
-		}
 		$pInputModelApiSecret->setValue($apiSecret);
 		$pInputModelApiKey->setSanitizeCallback(function ($apiKey) {
 			return $this->encrypteCredentials($apiKey);
 		});
 		$pInputModelApiSecret->setSanitizeCallback(function($password) use ($optionNameSecret) {
 			$password = $this->encrypteCredentials($password);
 			if (defined('ONOFFICE_CREDENTIALS_ENC_KEY') && empty($password)) {
-				
 				$password = $this->encrypteCredentials($this->checkPassword($password, $optionNameSecret));
+				update_option('ONOFFICE_CREDENTIALS_ENC_KEY',ONOFFICE_CREDENTIALS_ENC_KEY);
```

Comment:
The encryption key cannot be saved in the DB along with the encrypted secret. That would be pointless.